src/actions/index.js

```diff
@@ -177,22 +177,39 @@ export const deviceModaleRevokeClose = () => ({
 })
 
 export const fetchSessions = () => {
-  return (dispatch, getState) => {
+  return async (dispatch, getState) => {
     dispatch({ type: FETCH_SESSIONS })
-    cozyFetch('GET', '/data/io.cozy.sessions.logins/_all_docs?include_docs=true')
-    .then(response => {
-      const sessions = []
-      response.rows.map(row => sessions.push(row.doc))
-      dispatch({ type: FETCH_SESSIONS_SUCCESS, sessions })
-    })
-    .catch(() => {
+
+    const sessions = []
+    let responseSessions
+    let responseCurrentsSessionsIds
+
+    try {
+      // GET all the sessions
+      responseSessions = await cozyFetch('GET', '/data/io.cozy.sessions.logins/_all_docs?include_docs=true')
+      // Sort allSessions in an array
+      responseSessions.rows.map(row => sessions.push(row.doc))
+      // GET currents sessions id
+      responseCurrentsSessionsIds = await cozyFetch('GET', '/settings/sessions')
+    } catch (error) {
       dispatch({
         type: FETCH_SESSIONS_FAILURE,
         alert: {
           message: 'SessionsView.infos.server_error'
         }
       })
+    }
+
+    // Merge ID and Sessions to inject only currents sessions in the store
+    const currentsSessions = []
+
+    responseCurrentsSessionsIds.data.map((currentSessionId) => {
```

Comment:
I can probably use a shiny ES6 function here with .reduce or whatever, if you have it in mind !

Reply:
I suggest:

```js
currentsSessions.filter(
  session =>responseCurrentsSessionsIds.data.includes(session.session_id)
)
```

Reply:
:sparkles: :sparkles: Thx :)

Reply:
.includes doesn't work with an array of object :/ my solution isn't the more elegant, but it works, lets merge it ?

Reply:
Oh I didn't notice the `.id`, is it better? 

```
const currentSessions = sessions.filter( session => 
  for (let session of sessions) {
    return currentSessionId.id === session.session_id
  }
)
```

Reply:
We can keep it like, it's correct too 👍